### PR TITLE
Exclude PLCrashReporter, OCMock and OCHamcrest from additional warnings

### PR DESCRIPTION
**Pull request by MatkovIvan**
For suppress warnings from headers in the 3rd party libraries, mark they as "system headers" for compiler.

### CODE REVIEW

**Comment by jaelim-ms** [comment]
Can you also look at `base/updates` branch for MobileCenterDistribute project and 'base/push` branch for MobileCenterPush project? Crashes and Analytics projects changes will automatically go to those branches by merging but the projects won't get fixed if you don't do in their own branches.

File: MobileCenter/MobileCenter.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1016,11 +1016,19 @@
...
 				LIBRARY_SEARCH_PATHS = "\"$(SRCROOT)/../Vendor\"/**";
+				OTHER_CFLAGS = (
+					"$(inherited)",
+					"-iframework",
+					"\"$(SRCROOT)/../Vendor/OCHamcrest\"",
+					"-iframework",
+					"\"$(SRCROOT)/../Vendor/OHHTTPStubs\"",
+					"-isystem",
+					"\"$(SRCROOT)/../Vendor/OCMock\"",
+				);
```
Comment (TroubleMakerBen) [inline MobileCenter/MobileCenter.xcodeproj/project.pbxproj]: Please move those to the xcconfig-file. We want to utilize thos as much as possible.  The entry in the project file will then only just be:
```
OTHER_CFLAGS = (
					"$(inherited)"
);```

File: MobileCenter/MobileCenter.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1038,11 +1046,19 @@
...
```
Comment (TroubleMakerBen) [inline MobileCenter/MobileCenter.xcodeproj/project.pbxproj]: Same as above

File: MobileCenterAnalytics/MobileCenterAnalytics.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -657,11 +655,19 @@
...
 				LIBRARY_SEARCH_PATHS = "\"$(SRCROOT)/../Vendor\"/**";
+				OTHER_CFLAGS = (
+					"$(inherited)",
+					"-iframework",
+					"\"$(SRCROOT)/../Vendor/OCHamcrest\"",
+					"-iframework",
+					"\"$(SRCROOT)/../Vendor/OHHTTPStubs\"",
+					"-isystem",
+					"\"$(SRCROOT)/../Vendor/OCMock\"",
+				);
```
Comment (TroubleMakerBen) [inline MobileCenterAnalytics/MobileCenterAnalytics.xcodeproj/project.pbxproj]: Same as above

File: MobileCenterAnalytics/MobileCenterAnalytics.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -680,11 +686,19 @@
...
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
 				LIBRARY_SEARCH_PATHS = "\"$(SRCROOT)/../Vendor\"/**";
+				OTHER_CFLAGS = (
+					"$(inherited)",
+					"-iframework",
+					"\"$(SRCROOT)/../Vendor/OCHamcrest\"",
+					"-iframework",
+					"\"$(SRCROOT)/../Vendor/OHHTTPStubs\"",
+					"-isystem",
+					"\"$(SRCROOT)/../Vendor/OCMock\"",
```
Comment (TroubleMakerBen) [inline MobileCenterAnalytics/MobileCenterAnalytics.xcodeproj/project.pbxproj]: Same as above

File: MobileCenterCrashes/MobileCenterCrashes.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -859,9 +859,10 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				"ARCHS[sdk=iphonesimulator*]" = "$(MS_SIM_ARCHS)";
-				FRAMEWORK_SEARCH_PATHS = (
+				OTHER_CFLAGS = (
 					"$(inherited)",
-					"\"$(SRCROOT)/../Vendor\"/**",
+					"-iframework",
+					"\"$(SRCROOT)/../Vendor/PLCrashReporter\"",
```
Comment (TroubleMakerBen) [inline MobileCenterCrashes/MobileCenterCrashes.xcodeproj/project.pbxproj]: Same as above

**Comment by MatkovIvan** [comment]
@jaelim-ms I will create 2 PR to this branches after team accepted this one.

**Pull request merged**